compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -210,8 +212,8 @@ class ComposeScene internal constructor(
     private val coroutineScope = CoroutineScope(coroutineContext + job)
     // We use FlushCoroutineDispatcher for effectDispatcher not because we need `flush` for
     // LaunchEffect tasks, but because we need to know if it is idle (hasn't scheduled tasks)
-    private val effectDispatcher = FlushCoroutineDispatcher(coroutineScope)
-    private val recomposeDispatcher = FlushCoroutineDispatcher(coroutineScope)
+    private val effectDispatcher: CoroutineDispatcher = customEffectDispatcher ?: FlushCoroutineDispatcher(coroutineScope)
```

Comment:
@igordmn this is the change I think you can have concerns. 
It's needed only for tests.